react-native/react/pinentry/pinentry-render.js

```diff
@@ -1,30 +1,47 @@
 'use strict'
 /* @flow */
 
-import React from 'react'
-import BaseComponent from '../base-component'
-import { TextField, RaisedButton } from 'material-ui'
+import React, { Component } from '../base-react'
+import { TextField, RaisedButton, Checkbox } from 'material-ui'
 
-export default class PinentryRender extends BaseComponent {
+export default class PinentryRender extends Component {
   componentWillMount () {
     this.state = {
       passphrase: ''
     }
+    for (let feature in this.props.payload.features) {
+      if (this.props.payload.features[feature].hasOwnProperty('value')) {
+        this.state[feature] = this.props.payload.features[feature].value
+      } else {
+        console.error('We were passed a payload with no value!')
+      }
+    }
   }
 
   render () {
     return (
       <div>
-        <p>Please enter the Keybase passphrase for {this.props.user} (12+ characters)</p>
+        <p>{this.props.payload.prompt_text}</p>
         <TextField
           ref='passphrase'
           onChange={e => this.setState({passphrase: e.target.value})}
           floatingLabelText='Your passphrase'
           value={this.state.passphrase} />
 
+        {Object.keys(this.props.payload.features).map((feature, _) => {
+          return <Checkbox
```

Comment:
It's not clear to me this is that generalizable. The 'features' aren't just an array of checkboxes you can turn on / off. This is fine for now but I think the idea was that as features were added they were features of the pinentry as well (like a checkbox to show the text as you're typing instead of hiding it, etc)